src/main/java/uk/gov/hmcts/reform/unspec/service/flowstate/StateFlowEngine.java

```diff
@@ -70,81 +79,83 @@
     public StateFlow build() {
         return StateFlowBuilder.<FlowState.Main>flow(FLOW_NAME)
             .initial(DRAFT)
-                .transitionTo(PENDING_CASE_ISSUED).onlyIf(pendingCaseIssued)
-            .state(PENDING_CASE_ISSUED)
-                .transitionTo(PAYMENT_SUCCESSFUL).onlyIf(paymentSuccessful)
-                .transitionTo(PAYMENT_FAILED).onlyIf(paymentFailed)
-            .state(PAYMENT_FAILED)
-                .transitionTo(PAYMENT_SUCCESSFUL).onlyIf(paymentSuccessful)
-            .state(PAYMENT_SUCCESSFUL)
-                .transitionTo(AWAITING_CASE_NOTIFICATION).onlyIf(claimIssued)
-                .transitionTo(PROCEEDS_OFFLINE_UNREPRESENTED_DEFENDANT).onlyIf(respondent1NotRepresented)
+                .transitionTo(CLAIM_SUBMITTED).onlyIf(claimSubmitted)
+            .state(CLAIM_SUBMITTED)
+                .transitionTo(CLAIM_ISSUED_PAYMENT_SUCCESSFUL).onlyIf(paymentSuccessful)
+                .transitionTo(CLAIM_ISSUED_PAYMENT_FAILED).onlyIf(paymentFailed)
+            .state(CLAIM_ISSUED_PAYMENT_FAILED)
+                .transitionTo(CLAIM_ISSUED_PAYMENT_SUCCESSFUL).onlyIf(paymentSuccessful)
+            .state(CLAIM_ISSUED_PAYMENT_SUCCESSFUL)
+                .transitionTo(PENDING_CLAIM_ISSUED).onlyIf(pendingClaimIssued)
+                .transitionTo(PENDING_CLAIM_ISSUED_UNREPRESENTED_DEFENDANT).onlyIf(respondent1NotRepresented)
                 .transitionTo(PENDING_CLAIM_ISSUED_UNREGISTERED_DEFENDANT).onlyIf(respondent1OrgNotRegistered)
-            .state(AWAITING_CASE_NOTIFICATION)
-                .transitionTo(AWAITING_CASE_DETAILS_NOTIFICATION).onlyIf(claimNotified)
-                .transitionTo(CASE_PROCEEDS_IN_CASEMAN).onlyIf(caseProceedsInCaseman)
-                .transitionTo(CLAIM_DISMISSED_PAST_CLAIM_NOTIFICATION_DEADLINE).onlyIf(failToNotifyClaim)
-            .state(AWAITING_CASE_DETAILS_NOTIFICATION)
-                .transitionTo(CLAIM_ISSUED).onlyIf(claimDetailsNotified)
-                .transitionTo(CASE_PROCEEDS_IN_CASEMAN).onlyIf(caseProceedsInCaseman)
+            .state(PENDING_CLAIM_ISSUED)
+                .transitionTo(CLAIM_ISSUED).onlyIf(claimIssued)
+            .state(PENDING_CLAIM_ISSUED_UNREPRESENTED_DEFENDANT)
+                .transitionTo(TAKEN_OFFLINE_UNREPRESENTED_DEFENDANT).onlyIf(takenOfflineBySystem)
+            .state(PENDING_CLAIM_ISSUED_UNREGISTERED_DEFENDANT)
+                .transitionTo(TAKEN_OFFLINE_UNREGISTERED_DEFENDANT).onlyIf(takenOfflineBySystem)
+            .state(CLAIM_ISSUED)
+                .transitionTo(CLAIM_NOTIFIED).onlyIf(claimNotified)
+                .transitionTo(TAKEN_OFFLINE_BY_STAFF).onlyIf(takenOfflineByStaff)
+                .transitionTo(CLAIM_DISMISSED_PAST_CLAIM_NOTIFICATION_DEADLINE).onlyIf(pastClaimNotificationDeadline)
+            .state(CLAIM_NOTIFIED)
+                .transitionTo(CLAIM_DETAILS_NOTIFIED).onlyIf(claimDetailsNotified)
+                .transitionTo(TAKEN_OFFLINE_BY_STAFF).onlyIf(takenOfflineByStaff)
```

Comment:
Scenario: Claim taken offline after CLAIM_NOTIFIED. state flow engine in error! 

`Ambiguous transitions permitting state [MAIN.CLAIM_ISSUED] to move to more than one next states [MAIN.CLAIM_NOTIFIED,MAIN.TAKEN_OFFLINE_BY_STAFF].` 

Please add this test and it should pass.
```
 @Test
        void shouldReturnProceedsWithOfflineJourney_whenCaseDataIsCaseProceedsInCaseman() {
            CaseData caseData = CaseDataBuilder.builder().atStateClaimNotified()
                .claimProceedsInCaseman()
                .build();

            StateFlow stateFlow = stateFlowEngine.evaluate(caseData);

            assertThat(stateFlow.getState())
                .extracting(State::getName)
                .isNotNull()
                .isEqualTo(TAKEN_OFFLINE_BY_STAFF.fullName());
            assertThat(stateFlow.getStateHistory())
                .hasSize(7)
                .extracting(State::getName)
                .containsExactly(
                    DRAFT.fullName(), CLAIM_SUBMITTED.fullName(), CLAIM_ISSUED_PAYMENT_SUCCESSFUL.fullName(),
                    PENDING_CLAIM_ISSUED.fullName(), CLAIM_ISSUED.fullName(), CLAIM_NOTIFIED.fullName(), TAKEN_OFFLINE_BY_STAFF.fullName()
                );
        }
```